test.py

```diff
@@ -1,39 +1,262 @@
 #!/usr/bin/env python3
 
-from parse import *
+from unittest.mock import patch, mock_open
+import parse
 import logging
 import unittest
+import constants
+import csv
 
-class EncodingLineTest(unittest.TestCase):
-    def setUp(self):
-        logger = logging.getLogger()
-        logger.disabled = True
+# Configure logging
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
 
-    def assertError(self, string):
-        self.assertRaises(SystemExit, process_enc_line, string, 'rv_i')
-
-    def test_lui(self):
-        name, data = process_enc_line('lui     rd imm20 6..2=0x0D 1=1 0=1', 'rv_i')
+class TestProcessEncLine(unittest.TestCase):
+    @patch('builtins.open', new_callable=mock_open, read_data='lui rd 6..2=0x0D 1=1 0=1')
```

Comment:
@TraXIcoN here I would suggest moving the test data i.e., `lui rd 6..2=0x0D 1=1 0=1`, to a global variable or in the constants and referring that variable here, so that in the future if we need to change the test data, it could be done easily by modifying the assigned data to the variable. 
This will also add up to the cleaner code.